app/controllers/clients_controller.rb

```diff
@@ -5,32 +5,22 @@ class ClientsController < ApplicationController
   before_filter :check_authorized_access, except: [:index, :new]
  
   def index
-    @clients = Client.where("company_id = ?",current_user.company_id)
+    @clients = Client.where(company_id: current_user.company_id)
   end
 
   def new
     @client = Client.new
-    @address = @client.address
-    if @address.present?
-    else
-      @address = Address.new
-    end
+    @address = Address.new
+    @client.address = @address
   end
 
   def create
     @client = Client.new(client_params)
-    @client.company_id = current_user.company_id
-    @address = Address.new(address_params)
-    if @client.save
-      @address.client = @client
-      if @address.save
-        flash[:success] = 'Client and address saved properly'   
-        redirect_to clients_path
-      else
-        flash[:error] = "Address has some problem in save. #{@address.errors.full_messages.join(',')}"       
-        @client.destroy
-        render :action => 'new'
-      end
+    @client.company_id = current_user.company_id    
+    @address = @client.address
+    if @client.save      
+      flash[:success] = 'Client and address saved properly'   
+      redirect_to clients_path      
     else
       flash[:error] = "Client has some problem in save. #{@client.errors.full_messages.join(',')}"
```

Comment:
replace `Client has some problem in save` with `Problem while saving client details`